### PR TITLE
Add missed available transformations to Migration Doc

### PR DESCRIPTION
**Pull request by maurogeorge**

### CODE REVIEW

File: activerecord/lib/active_record/migration.rb [hunk 1]
```diff
@@ -106,17 +106,16 @@ def initialize
   #
   # == Available transformations
   #
+  # === Creation
+  #
+  # * <tt>create_join_table(table_1, table_2, options)</tt>: Creates a new join
```
Comment (repinel) [inline activerecord/lib/active_record/migration.rb]: :scissors: `new` ?

File: activerecord/lib/active_record/migration.rb [hunk 2]
```diff
@@ -106,17 +106,16 @@ def initialize
...
+  #   table with the name created using the lexical order of the first two
```
Comment (repinel) [inline activerecord/lib/active_record/migration.rb]: How about `table having its name as the lexical (...)`?
Should we also note that the table object will be provided to a block, just like `create_table`?

File: activerecord/lib/active_record/migration.rb [hunk 3]
```diff
@@ -127,24 +126,58 @@ def initialize
...
-  #   named +column_name+ from the table called +table_name+.
+  # * <tt>add_foreign_key(from_table, to_table, options)</tt>: Adds a new
+  #   foreign key. +from_table+ is the table with the key column, +to_table+ contains
+  #   the referenced primary key.
   # * <tt>add_index(table_name, column_names, options)</tt>: Adds a new index
   #   with the name of the column. Other options include
   #   <tt>:name</tt>, <tt>:unique</tt> (e.g.
   #   <tt>{ name: 'users_name_index', unique: true }</tt>) and <tt>:order</tt>
   #   (e.g. <tt>{ order: { name: :desc } }</tt>).
+  # * <tt>add_reference(table_name, reference_name)</tt>: Adds a new column
```
Comment (repinel) [inline activerecord/lib/active_record/migration.rb]: Should we add `ActiveRecord::ConnectionAdapters::SchemaStatements#add_reference` reference back here?

`change_column_null` has one to `SchemaStatements`...

File: activerecord/lib/active_record/migration.rb [hunk 4]
```diff
@@ -127,24 +126,58 @@ def initialize
...
+  #   a column but keeps the type and content.
+  # * <tt>rename_index(table_name, old_name, new_name)</tt>: Renames an index.
+  # * <tt>rename_table(old_name, new_name)</tt>: Renames the table called +old_name+
+  #   to +new_name+.
+  #
+  # === Deletion
+  #
+  # * <tt>drop_table(name)</tt>: Drops the table called +name+.
+  # * <tt>drop_join_table(table_1, table_2, options)</tt>: Drops the join table
+  #   specified by the given arguments.
```
Comment (repinel) [inline activerecord/lib/active_record/migration.rb]: Worth mentioning the lexical order of the arguments?

**Comment by repinel** [comment]
I liked the idea of having the doc categorized by the nature of the methods (e.g. `Creation`). :smile:

File: activerecord/lib/active_record/migration.rb (hunk 3 above)
Comment (maurogeorge) [inline activerecord/lib/active_record/migration.rb]: I added a link only on the more "complex" methods. In fact I am thinking about add a link in each method, but maybe a lot of links became confusing on the Docs.

**Comment by maurogeorge** [comment]
@repinel thanks for the review :star2: 
I updated with your suggestions

**Comment by maurogeorge** [comment]
Thanks for the merge :heart:

**Pull request merged**